File: app/lib/models/institution_administrator_position.dart
class InstitutionAdministratorPosition {
  final int id;
  final String createdAt;
  final String updatedAt;
  final String name;

  const InstitutionAdministratorPosition({
    required this.id,
    required this.createdAt,
    required this.updatedAt,
    required this.name,
  });

  factory InstitutionAdministratorPosition.fromJson(
    Map<String, dynamic> json,
  ) => InstitutionAdministratorPosition(
    id: json['id'] as int,
    createdAt: json['createdAt'] as String,
    updatedAt: json['updatedAt'] as String,
    name: json['name'] as String,
  );
}

File: backend/internal/model/institution_administrator_position.go
package model

import (
	"gorm.io/gorm"
	"time"
)

type InstitutionAdministratorPosition struct {
	ID        uint16 `gorm:"primaryKey"`
	CreatedAt time.Time
	UpdatedAt time.Time
	DeletedAt gorm.DeletedAt
	Name      string `gorm:"type:varchar(200);unique;check:length(trim(name)) >= 4"`
	// 1. Can't remove position if there are at least one person with it
	// 2. one-to-many (position-to-administrator)
	InstitutionAdministrators []InstitutionAdministrator `gorm:"foreignKey:PositionID;references:ID;constraint:OnDelete:restrict,OnUpdate:restrict"`
}
